### PR TITLE
DSND-827: Company Links Consumer - Process ResourceChangedData delete…

### PR DESCRIPTION
**Pull request by sameerinfo2003**
https://companieshouse.atlassian.net/browse/DSND-827

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/company/links/processor/InsolvencyStreamProcessor.java [hunk 1]
```diff
@@ -52,6 +51,35 @@ public void processDelete(Message<ResourceChangedData> resourceChangedMessage) {
...
+        var links = data.getLinks();
+
+        if (links == null) {
+            logger.trace(String.format("Company profile with company number %s,"
+                            + " does not contain insolvency links, will not perform patch",
+                    companyNumber));
+            return;
+        }
+
+        data.setLinks(null);
```
Comment (gseregni-ch) [inline src/main/java/uk/gov/companieshouse/company/links/processor/InsolvencyStreamProcessor.java:70]: We need only to remove the insolvency link and not all the links. Please address this.

**Review by gseregni-ch — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/companieshouse/company/links/processor/InsolvencyStreamProcessor.java (hunk 1 above)
Comment (sameerinfo2003) [reply src/main/java/uk/gov/companieshouse/company/links/processor/InsolvencyStreamProcessor.java]: changes done

File: src/main/java/uk/gov/companieshouse/company/links/processor/InsolvencyStreamProcessor.java [hunk 2]
```diff
@@ -52,6 +51,36 @@ public void processDelete(Message<ResourceChangedData> resourceChangedMessage) {
...
+        var links = data.getLinks();
+
+        if (links != null && links.getInsolvency() == null) {
+            logger.trace(String.format("Company profile with company number %s,"
+                            + " does not contain insolvency links, will not perform patch",
+                    companyNumber));
+            return;
+        }
+
+        data.setHasInsolvencyHistory(false);
```
Comment (gseregni-ch) [inline src/main/java/uk/gov/companieshouse/company/links/processor/InsolvencyStreamProcessor.java:70]: Sorry Sameer. Differently to what we discussed, please do not set the InsolvencyHistory to false. I re-discussed this with Zahid. Please change also the related changes for the tests.

**Review by gseregni-ch — CHANGES_REQUESTED** [review]

**Review by gseregni-ch — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/company/links/processor/InsolvencyStreamProcessor.java (hunk 2 above)
Comment (sameerinfo2003) [reply src/main/java/uk/gov/companieshouse/company/links/processor/InsolvencyStreamProcessor.java]: code change done

**Review by zahidm3 — APPROVED** [review]

**Pull request merged**